src/sage/combinat/skew_partition.py

```diff
@@ -1490,7 +1516,15 @@ def _element_constructor_(self, skp):
             sage: S = SkewPartitions()
             sage: S([[3,1], [1]])
             [3, 1] / [1]
+
+        TESTS::
+
+            sage: S = SkewPartitions()
+            sage: S(Partition([3,2]))
+            [3, 2] / []
         """
+        if skp in _Partitions:
+            skp = SkewPartition(skp)
```

Comment:
```suggestion
        if skp in _Partitions:
            skp = [_Partitions(skp), _Partitions([])]
```
Less indirection (so faster) and will result in the correct parent. Please add an analogous test with a different parent (such as `SkewPartitions(4)`) and make sure it has the correct output.